File: E-TicketingBackend/E-TicketingBackend/Model/RequestDTO.cs
﻿using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.ComponentModel.DataAnnotations;

namespace E_TicketingBackend.Model
{
    public class RequestDTO
    {

        public string? trainCode { get; set; }
        public string? nic { get; set; }
        public UserDTO? userDto { get; set; }
        public TicketDTO? ticketDto { get; set; }
        public ScheduleDTO? scheduleDTO { get; set; }
        public TrainDTO? trainDTO { get; set; }
    }
}

File: mobileClient/app/src/main/java/com/sliit/eticket/dto/RequestDto.java
package com.sliit.eticket.dto;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/*
 *  This class is use to create request object
 * */
public class RequestDto implements Serializable {

    private  String nic;
    @SerializedName("ticketDto")
    private TicketDto ticketDto;
    @SerializedName("userDTO")
    private AppUserDto userDTO;
    @SerializedName("scheduleDTO")
    private ScheduleDto scheduleDTO;
    @SerializedName("trainDTO")
    private TrainDto trainDTO;

    public RequestDto() {

    }

    public String getNic() {
        return nic;
    }

    public void setNic(String nic) {
        this.nic = nic;
    }

    public TicketDto getTicketDto() {
        return ticketDto;
    }

    public void setTicketDto(TicketDto ticketDto) {
        this.ticketDto = ticketDto;
    }

    public AppUserDto getUserDTO() {
        return userDTO;
    }

    public void setUserDTO(AppUserDto userDTO) {
        this.userDTO = userDTO;
    }

    public ScheduleDto getScheduleDTO() {
        return scheduleDTO;
    }

    public void setScheduleDTO(ScheduleDto scheduleDTO) {
        this.scheduleDTO = scheduleDTO;
    }

    public TrainDto getTrainDTO() {
        return trainDTO;
    }

    public void setTrainDTO(TrainDto trainDTO) {
        this.trainDTO = trainDTO;
    }
}
